agent/logger/log_test.go

```diff
@@ -332,46 +332,109 @@ func TestSeelogConfig_FileLevelDefault(t *testing.T) {
+		logDriver                string
+		loglevel                 string
+		loglevelInstance         string
+		expectedLoglevel         string
+		expectedLoglevelInstance string
+	}{
+		{
+			name:                     "nothing set",
+			logDriver:                "",
+			loglevel:                 "",
+			loglevelInstance:         "",
+			expectedLoglevel:         "info",
+			expectedLoglevelInstance: "info",
+		},
+		{
+			name:                     "only loglevel",
+			logDriver:                "",
+			loglevel:                 "debug",
+			loglevelInstance:         "",
+			expectedLoglevel:         "debug",
+			expectedLoglevelInstance: "debug",
+		},
+		{
+			name:                     "only on instance",
+			logDriver:                "",
+			loglevel:                 "",
+			loglevelInstance:         "debug",
+			expectedLoglevel:         "info",
+			expectedLoglevelInstance: "debug",
+		},
+		{
+			name:                     "both levels no driver",
+			logDriver:                "",
+			loglevel:                 "warn",
+			loglevelInstance:         "crit",
+			expectedLoglevel:         "warn",
+			expectedLoglevelInstance: "critical",
+		},
+		{
+			name:                     "loglevel and driver",
+			logDriver:                "journald",
+			loglevel:                 "debug",
+			loglevelInstance:         "",
+			expectedLoglevel:         "debug",
+			expectedLoglevelInstance: "off",
+		},
+		{
+			name:                     "loglevel on instance and driver",
+			logDriver:                "journald",
+			loglevel:                 "",
+			loglevelInstance:         "debug",
+			expectedLoglevel:         "info",
+			expectedLoglevelInstance: "debug",
+		},
+		{
+			name:                     "both levels and driver",
+			logDriver:                "journald",
+			loglevel:                 "warn",
+			loglevelInstance:         "debug",
+			expectedLoglevel:         "warn",
+			expectedLoglevelInstance: "debug",
+		},
+		{
+			name:                     "only driver",
+			logDriver:                "journald",
+			loglevel:                 "",
+			loglevelInstance:         "",
+			expectedLoglevel:         "info",
+			expectedLoglevelInstance: "off",
+		},
 	}
-	SetLevel(os.Getenv(LOGLEVEL_ENV_VAR), os.Getenv(LOGLEVEL_ON_INSTANCE_ENV_VAR))
-	require.Equal(t, "debug", Config.driverLevel)
-	require.Equal(t, "critical", Config.instanceLevel)
 
-	c := seelogConfig()
-	require.Equal(t, `
-<seelog type="asyncloop">
-	<outputs formatid="logfmt">
-		<filter levels="debug,info,warn,error,critical">
-			<console />
-		</filter>
-		<filter levels="critical">
-			<rollingfile filename="foo.log" type="date"
-			 datepattern="2006-01-02-15" archivetype="none" maxrolls="24" />
-		</filter>
-	</outputs>
-	<formats>
-		<format id="logfmt" format="%EcsAgentLogfmt" />
-		<format id="json" format="%EcsAgentJson" />
-		<format id="windows" format="%Msg" />
-	</formats>
-</seelog>`, c)
+	for _, test := range testcases {
+		t.Run(test.name, func(t *testing.T) {
+			resetEnv()
```

Comment:
why do you need to reset these here? setenv will override them anyway. instead i think you should have `defer resetEnv()` to make sure the envs are unset when test finished